### PR TITLE
FS 950 document upload

### PR DESCRIPTION
**Pull request by ramsharma-prog**
1. Reusable HTML template to upload documents.
2. Import macro: {% from "components/upload_documents.html" **import uploadDocuments** as **upload** %}.
3. Call upload macro  {{ upload (**title, url**) }}

Screenshot 2022-07-01 at 15 38 40

### CODE REVIEW

File: app/assess/routes.py [hunk 1]
```diff
@@ -138,3 +140,47 @@ def fund_round(fund_id: str, round_id: str):
...
+            }
+        ],
+    }
+
+    json = uploaded_file_json["questions"][0]["fields"][0]
+
+    return render_template(
+        "macros/example_upload_documents.html",
+        title=json["title"],
+        url=json["answer"],
```
Comment (SentientHousePlant) [inline app/assess/routes.py:185]: 💯 🔧 Great stuff Ram and a rapid fix, although we may want the page to take a list so we can render multiple files at once :)

File: app/assess/templates/macros/example_upload_documents.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{% extends "base.html" %}
+{% from "macros/upload_documents.html" import uploadDocuments as upload %}
+
+{% block content %}
+{{ upload(title, url) }}
```
Comment (SentientHousePlant) [inline app/assess/templates/macros/example_upload_documents.html:5]: 🤔 You could do a FOR loop there to render multiple files :)

File: requirements.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ certifi==2021.10.8
 cffi==1.15.0
 cfgv==3.3.1
 charset-normalizer==2.0.11
-chromedriver-py==100.0.4896.60
+chromedriver-py==102.0.5005.61
```
Comment (SentientHousePlant) [inline requirements.txt:11]: 📌 Consider creating a BAU to change this to web driver-manager.

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]
just some small changes but well done for rapidly updating it...almost there :)

File: requirements.txt (hunk 3 above)
Comment (ramsharma-prog) [reply requirements.txt:11]: Will do another PR for it

**Review by gio-karunakaran — COMMENTED** [review]
just 1 comment from me but looks great otherwise! 🙌

File: app/assess/templates/macros/upload_documents.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% macro uploadDocuments(title,url) %}
+
+<div class="govuk-grid-row">
+
+    <div class="govuk-grid-column-one-third upload-file">
+        <h1 class="govuk-caption-xl upload-file-title">{{title}}</h1>
+        <a class="govuk-link govuk-link--no-underline" href="{{url}}">
+            {{title}}
+        </a>
+    </div>
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply app/assess/templates/macros/upload_documents.html:10]: Its the heading & <a> tag is a link as we are not sure about the design so I added title as a heading & a title as a  link for now 😄

[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply app/assess/templates/macros/upload_documents.html:10]: See the screenshot please. More than happy to discuss 😄

[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply app/assess/templates/macros/upload_documents.html:10]: ahh ok that make sense!

**Review by gio-karunakaran — APPROVED** [review]
great work! 💯

**Comment by ramsharma-prog** [comment]
> Looks good, although should the whole box function as a link?

> I will see if I can implement thumbnail to it 🙂

**Comment by ramsharma-prog** [comment]
> > Looks good, although should the whole box function as a link?
> 
> > I will see if I can implement thumbnail to it 🙂


> wrapped it in `**a** `tag

File: app/assess/routes.py [hunk 5]
```diff
@@ -179,6 +179,36 @@ def fund_round(fund_id: str, round_id: str):
...
+                ]
+            }
+        ]
+    }
+                    
+    json_fields = uploaded_file_json["questions"][0]["fields"][0]
+    attachment_file_json = QuestionField.from_json(json_fields)
+
+    return render_template(
+        "macros/example_upload_documents.html", file=attachment_file_json)
```
Comment (SentientHousePlant) [inline app/assess/routes.py:210]: ⛏️ Change to file_metadata

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]

File: app/assess/routes.py [hunk 6]
```diff
@@ -198,16 +197,18 @@ def upload_documents():
...
     }
-                    
+
     json_fields = uploaded_file_json["questions"][0]["fields"][0]
-    attachment_file_json = QuestionField.from_json(json_fields)
+    file_metadata = QuestionField.from_json(json_fields)
 
     return render_template(
-        "macros/example_upload_documents.html", file=attachment_file_json)         
+        "macros/example_upload_documents.html", file=file_metadata
```
Comment (SentientHousePlant) [inline app/assess/routes.py:209]: 🔧 Change name in template too :) sorry!! 😢

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]

**Review by SentientHousePlant — APPROVED** [review]

**Pull request merged**